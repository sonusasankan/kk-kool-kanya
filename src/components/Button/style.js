import styled from 'styled-components';
import { colors } from '../../themes';

export const StyledButton = styled.button`
    background: ${props => (props.primary ? `${colors.primary}` : 'none')};
    color: ${props => (props.primary ? '#fff' : '#703db2')};
    font-size: 1em;
    margin: ${props => (props.isAlignRight ? '0' : '')};
    padding:${props => (props.isAlignRight ? '0' : '0.50em 1.5em')};
    min-height: 50px;
    border: ${props => (props.hasBorder ? '2px solid #703db2' : 'none')}
    border-radius:${props => (props.curved ? '20px' : '3px')};
    cursor: pointer;
    transition: all 0.25s ease-in-out;
    &:hover{
    background-color: ${props => (props.primary ? `${colors.primaryDark}` : `${colors.primary}`)};
    color: #fff;
    border-color: ${props => (props.primary ? `${colors.primaryDark}` : `${colors.primary}`)};
    }
    &.kk-has-icon{
    &.hovered{
        background: transparent;
        color: ${colors.primary} 
        .icon{
        transition: all 0.50s ease-in-out;
        opacity: 0;
        transform: translateX(20px) scale(1.5);
        }
    }
    &.leave{
        .icon{
        transition: all 0.50s ease-in-out;
        opacity: 1;
        transform: translateX(0px);
        }
    }
    .icon{
        vertical-align: middle;
        width: 20px;
    }
    }
    &.curved{
        border-radius: 50px;
    }
    &.no-bg-hover{
        &:hover{
            background-color: transparent;
            color: ${colors.primary} 
        }
    }
`;
