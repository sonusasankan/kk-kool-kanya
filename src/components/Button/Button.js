import React, { Component } from "react";
import PropTypes from "prop-types";
import classnames from 'classnames';

import {StyledButton} from './style';


// all states for button
const STATUS = {
  HOVERED: "hovered",
  NORMAL: "normal",
  CLICKED: "clicked",
  LEAVE: "leave"
};

class Button extends Component {
  constructor(props) {
    super(props);

    //default state of button will be normal
    this.state = {
      class: STATUS.NORMAL
    };
  }

  _handleClick = () => {
    this.setState({ class: STATUS.CLICKED });
    this.props.click();
  };

  //hovered stated for button
  _onMouseEnter = () => {
    this.setState({ class: STATUS.HOVERED });
  };

  
  _onMouseLeave = () => {
    this.setState({ class: STATUS.LEAVE });
  };

  /* Button component*/
  
  render() {
    const { primary, hasBorder, label, icon, isAlignRight, children, click } = this.props;
    return (
      <StyledButton
        name="button"
        hasBorder={hasBorder}
        isAlignRight={isAlignRight}
        primary={primary}
        className={classnames(icon != null ? "kk-has-icon"+" "+this.state.class : this.state.class,  this.props.styleName)}
        onClick={this._handleClick}
        onMouseEnter={this._onMouseEnter}
        onMouseLeave={this._onMouseLeave}
      >
        <span>{label}</span>
        {icon}
        {children}
      </StyledButton>
    );
  }
}

Button.defaultProps = {
  primary: true,
  hasBorder: false,
  label: "",
  icon: null,
  isAlignRight: false
};

Button.propTypes = {
  label: PropTypes.string.isRequired,
  primary: PropTypes.bool,
  hasBorder: PropTypes.bool,
  isAlignRight: PropTypes.bool,
  styleName: PropTypes.string,
  click: PropTypes.func.isRequired
};

export { Button };
