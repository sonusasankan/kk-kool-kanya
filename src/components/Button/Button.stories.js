import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
// import { linkTo } from '@storybook/addon-links';
import { withKnobs, text } from '@storybook/addon-knobs';
import { FiArrowRight } from 'react-icons/fi';
import { withNotes } from '../../docs-addon';

import { Button } from './Button';

storiesOf('Button', module)
  .addDecorator(withKnobs)
  .addDecorator(withNotes)
  .add('Primary', () => (
    <Button
      label={text('label', 'Primary button')}
      onClick={action('clicked')}
    />
  ), { notes: JSON.stringify(Button.__docgenInfo, null, 10) })
  .add('Curved', () => (
    <Button
      styleName="curved"
      label={text('label', 'Curved button')}
      onClick={action('clicked')}
    />
  ), { notes: JSON.stringify(Button.__docgenInfo, null, 10) })
  .add('Secondary', () => (
    <Button
      primary={false}
      hasBorder
      label={text('label', 'Secondary button')}
    />
  ))
  .add('Has Icon', () => (
    <Button
      primary={false}
      label={text('label', 'View all')}
      icon={<FiArrowRight className="icon" />}
    />
  ));
