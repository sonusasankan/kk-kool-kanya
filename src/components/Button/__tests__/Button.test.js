import React from 'react';
import renderer from 'react-test-renderer';

import { shallow } from 'enzyme';
import { Button } from '../Button';


describe('Button', () => {
  it('should render correctly in "debug" mode', () => {
    const button = renderer.create(<Button label="Primary button" />);

    let tree = button.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onClick();
    // re-rendering
    tree = button.toJSON();
    expect(tree).toMatchSnapshot();

    // manually trigger the callback
    tree.props.onMouseLeave();
    // re-rendering
    tree = button.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
