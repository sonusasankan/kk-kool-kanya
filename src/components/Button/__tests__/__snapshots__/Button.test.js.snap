// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly in "debug" mode 1`] = `
<button
  className="normal sc-bdVaJa kMeujv"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Primary button
</button>
`;

exports[`Button should render correctly in "debug" mode 2`] = `
<button
  className="clicked sc-bdVaJa kMeujv"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Primary button
</button>
`;

exports[`Button should render correctly in "debug" mode 3`] = `
<button
  className="leave sc-bdVaJa kMeujv"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Primary button
</button>
`;
